pkg/kademlia/kademlia_test.go

```diff
@@ -450,6 +488,39 @@ func newTestServer(nn []*pb.Node) (*grpc.Server, *mockNodesServer) {
 	return grpcServer, mn
 }
 
+// TestRandomIds makes sure finds a random node ID is within a range (start..end]
+func TestRandomIds(t *testing.T) {
+	for x := 0; x < 1000; x++ {
```

Comment:
how long does it take to run this test 1000 times? I wonder if there is a better way to test this

Reply:
One thought I had would be selecting a predefined `start` and `end` rather than generating them randomly. And I'd select them so that there are only like 5-10 possibilities for random IDs between them. That way you could iterate fewer times and test more exhaustively.

It still isn't optimal but I am not sure how to exhaustively test something like this :/

Reply:
it take a fraction of a second.  it'd be prohibitive to test exhaustively, but I've run it several million times now.  its true that this is not a unit test, its a more of a fuzz / property-based test.